deploy/examples/create-external-cluster-resources.py

```diff
@@ -1074,6 +1086,10 @@ def create_rgw_admin_ops_user(self):
             "buckets=*;users=*;usage=read;metadata=read;zone=read",
             "--rgw-realm",
             self._arg_parser.rgw_realm_name,
+            "--rgw-zone",
+            self._arg_parser.rgw_zone_name,
```

Comment:
If no zone or zonegroup are specified, they will get the default of the empty string?

Reply:
Suggest re-ordering these to be realm, then zonegroup, then zone, since that is the order of hierarchy.

Reply:
Yaa, it will be "", I think that mean default

But will confirm

Reply:
@thotz 
What you will suggest doing here?
Either give default or live as empty?

Reply:
IMO giving default might be the right approach but only for zone and zonegroup. About empty strings I am not so sure, @cbodley ??

Reply:
I played around with some toy clusters with multisite enabled and not enabled and here's what I observed:

If a realm has been is created there is a default realm. This is different than the realm named `default` and the `realm-id` of this realm can be accessed via `radosgw-admin realm get-default`. 

If no realms have been created there is no default realm set, but running `radosgw-admin user create` with `--rgw-realm=default --rgw-zonegroup=default --rgw-zone=default`  is the same result as not having those flags for user creation.

Since the default realm is not the same as the realm named `default` in a multisite scenario, if the realm/zg/zone flags are not passed in or are set to empty values the users are created in the default realm but not in the realm `default`.
To be clear here is an example:
Say  2 rgw users U1, and U2 are created on a Ceph cluster. 
Then realm1 is configured (with a zg and zone) with users U3 and U4 created in that realm. Since this realm is the first created it becomes the default realm.
Then a realm2 is configured with users U5 and U6 created in that realm.

Users U1 and U2 live in a realm called `default`.
If a new user U7 is created without multisite flags passed into the `user create` command it is created in the default realm, realm1.
If a new user U8 is created with multisite flags set to empty `--rgw-realm= --rgw-zonegroup= --rgw-zone=` passed into the `user create` command it is created in the default realm, realm1.
If a new user U9 is created with multisite flags set to the string default `--rgw-realm=default --rgw-zonegroup=default --rgw-zone=default` passed into the `user create` command it is created in the realm named `default` that contains users U1 and U2.

I'm not exactly sure what the right course of action is but I wanted to give everyone in this discussion this background information.

Reply:
We just care about U7 and U8 for now, as I see they are the same so no need to worry, 

If a new user U7 is created without multisite flags passed into the user create command it is created in the default realm, realm1.
If a new user U8 is created with multisite flags set to empty --rgw-realm= --rgw-zonegroup= --rgw-zone= passed into the user create command it is created in the default realm, realm1.

Thanks for the great explanation @alimaredia